### ISSUE TITLE
Support dask 'auto' chunking

### ISSUE THREAD

**Issue by maawoo** [opendatacube/odc-stac#123]
In addition to #81, I think it would be great if it's possible to select `'auto'` for chunks as described [here](https://docs.dask.org/en/latest/array-chunks.html#automatic-chunking). 

This can work quite well in combination with changing the default chunk size in dask. Here is an example with stackstac:

<img width="1311" alt="image" src="https://github.com/opendatacube/odc-stac/assets/56583917/eafa1770-391a-446c-9b41-022cfd9754da">

This is the current error message when trying to use 'auto' in `odc-stac`:
```python
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[10], line 1
----> 1 ds2 = odc.stac.load(items=s1_rtc_items, chunks={'time': -1, 'x': 'auto', 'y': 'auto'}, 
      2                     crs='EPSG:4326', resolution=0.0002, dytpe='float32', resampling='bilinear')

File ~/micromamba/envs/jupy_env/lib/python3.11/site-packages/odc/stac/_load.py:497, in load(items, bands, groupby, resampling, dtype, chunks, pool, crs, resolution, anchor, geobox, bbox, lon, lat, x, y, like, geopolygon, progress, fail_on_error, stac_cfg, patch_url, preserve_original_order, **kw)
    494     raise ValueError("Failed to auto-guess CRS/resolution.")
    496 if chunks is not None:
--> 497     chunk_shape = _resolve_chunk_shape(gbox, chunks)
    498 else:
    499     chunk_shape = _resolve_chunk_shape(
    500         gbox,
    501         {dim: DEFAULT_CHUNK_FOR_LOAD for dim in gbox.dimensions},
    502     )

File ~/micromamba/envs/jupy_env/lib/python3.11/site-packages/odc/stac/_load.py:861, in _resolve_chunk_shape(gbox, chunks)
    858         return sz
    859     return chunk
--> 861 ny, nx = [
    862     _norm_dim(chunks.get(dim, chunks.get(fallback_dim, -1)), n)
    863     for dim, fallback_dim, n in zip(gbox.dimensions, ["y", "x"], gbox.shape.yx)
    864 ]
    865 return ny, nx

File ~/micromamba/envs/jupy_env/lib/python3.11/site-packages/odc/stac/_load.py:862, in <listcomp>(.0)
    858         return sz
    859     return chunk
    861 ny, nx = [
--> 862     _norm_dim(chunks.get(dim, chunks.get(fallback_dim, -1)), n)
    863     for dim, fallback_dim, n in zip(gbox.dimensions, ["y", "x"], gbox.shape.yx)
    864 ]
    865 return ny, nx

File ~/micromamba/envs/jupy_env/lib/python3.11/site-packages/odc/stac/_load.py:857, in _resolve_chunk_shape.<locals>._norm_dim(chunk, sz)
    856 def _norm_dim(chunk: int, sz: int) -> int:
--> 857     if chunk < 0 or chunk > sz:
    858         return sz
    859     return chunk

TypeError: '<' not supported between instances of 'str' and 'int'
```

**Comment by Kirill888** [comment]
problem with `auto` is that it means different things to different people. From your example `auto` in `stackstac` means "choose chunk size such that it's closest to a certain byte size". But given that we are loading data from some storage location, users might interpret `auto` as "align chunks to tiling regime used in storage", which is much more involved process and is not generally possible without peeking into the data files first.

In short: interpreting `auto` as "pick chunk size based on `dtype` and some dask configuration about preferred chunk size in bytes" is certainly possible and won't be too hard to implement, most work is in type annotation, but we need to carefully document this feature to avoid confusion with "align to storage chunks" assumption users might make.

**Comment by clausmichele** [comment]
@Kirill888 ~~I would re-open this issue, since apparently the "auto" doesn't work well with the temporal dimension!~~

Edit: solution found below, but I leave the sample because someone else might try the same.

minimal example:
```python
import pystac_client
import json
import odc.stac

url = "https://earth-search.aws.element84.com/v1/collections/sentinel-2-l2a/items/S2A_32TPS_20230602_0_L2A"
stac_api = pystac_client.stac_api_io.StacApiIO()
stac_dict = json.loads(stac_api.read_text(url))
items = stac_api.stac_object_from_dict(stac_dict)
ds_stac = odc.stac.load([items],chunks="auto")
```

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[5], line 9
      7 stac_dict = json.loads(stac_api.read_text(url))
      8 items = stac_api.stac_object_from_dict(stac_dict)
----> 9 ds_stac = odc.stac.load([items],chunks="auto")

File ~/anaconda3/envs/stac_310/lib/python3.10/site-packages/odc/stac/_load.py:550, in load(items, bands, groupby, resampling, dtype, chunks, pool, crs, resolution, anchor, geobox, bbox, lon, lat, x, y, like, geopolygon, progress, fail_on_error, stac_cfg, patch_url, preserve_original_order, **kw)
    547 tss = _extract_timestamps(ndeepmap(2, lambda idx: _parsed[idx], _grouped_idx))
    549 if chunks is not None:
--> 550     chunk_shape = _resolve_chunk_shape(len(tss), gbox, chunks, dtype)
    551 else:
    552     chunk_shape = _resolve_chunk_shape(
    553         len(tss),
    554         gbox,
    555         {dim: DEFAULT_CHUNK_FOR_LOAD for dim in gbox.dimensions},
    556         dtype,
    557     )

File ~/anaconda3/envs/stac_310/lib/python3.10/site-packages/odc/stac/_load.py:884, in _resolve_chunk_shape(nt, gbox, chunks, dtype)
    881 def _resolve_chunk_shape(
    882     nt: int, gbox: GeoBox, chunks: Dict[str, int | Literal["auto"]], dtype: Any
    883 ) -> Tuple[int, int, int]:
--> 884     tt = chunks.get("time", 1)
    885     ty, tx = (
    886         chunks.get(dim, chunks.get(fallback_dim, -1))
    887         for dim, fallback_dim in zip(gbox.dimensions, ["y", "x"])
    888     )
    889     nt, ny, nx = (
    890         ch[0]
    891         for ch in normalize_chunks((tt, ty, tx), (nt, *gbox.shape.yx), dtype=dtype)
    892     )

AttributeError: 'str' object has no attribute 'get'
```

Edit: probably I misunderstood how to use this parameter, this line works instead:
```python
ds_stac = odc.stac.load([items],chunks={"time": 1, "x": "auto", "y": "auto"})
```

But is it possible to set "auto" for all the available dimensions without knowing them in advance?
